### ISSUE TITLE
Build broken?

### ISSUE THREAD

**Issue by TommyMurphyTM1234** [gnu-mcu-eclipse/openocd-build#3]
I followed the usual build instructions here:

https://gnu-mcu-eclipse.github.io/openocd/build-procedure/

but got this when building on Ubuntu 16.04 64 bit:

```
bash ~/Downloads/openocd-build.git/scripts/build.sh --win32

...

libtool: compile:  i686-w64-mingw32-gcc -DHAVE_CONFIG_H -I. -I/Host/Work/openocd/openocd.git -D__USE_MINGW_ANSI_STDIO -I/Host/Work/openocd/openocd.git/src -I./src -I/Host/Work/openocd/openocd.git/src/helper -DPKGDATADIR=\"\" -DBINDIR=\"bin\" -I/Host/Work/openocd/openocd.git/jimtcl -I./jimtcl -isystem /tmp/install/win32/include/libusb-1.0 -I/tmp/install/win32/include/libusb -I/tmp/install/win32/include/libftdi1 -I/tmp/install/win32/include/libusb-1.0 -I/tmp/install/win32/include/hidapi -I./src/jtag/drivers/libjaylink/libjaylink -I/Host/Work/openocd/openocd.git/src/jtag/drivers/libjaylink -Wall -Wstrict-prototypes -Wformat-security -Wshadow -ffunction-sections -fdata-sections -m32 -pipe -Wno-pointer-to-int-cast -MT src/jtag/drivers/src_jtag_drivers_libocdjtagdrivers_la-remote_bitbang.lo -MD -MP -MF src/jtag/drivers/.deps/src_jtag_drivers_libocdjtagdrivers_la-remote_bitbang.Tpo -c /Host/Work/openocd/openocd.git/src/jtag/drivers/remote_bitbang.c -o src/jtag/drivers/src_jtag_drivers_libocdjtagdrivers_la-remote_bitbang.o
/Host/Work/openocd/openocd.git/src/jtag/drivers/remote_bitbang.c: In function 'remote_bitbang_fill_buf':
/Host/Work/openocd/openocd.git/src/jtag/drivers/remote_bitbang.c:63:2: warning: implicit declaration of function 'fcntl'; did you mean 'fcvt'? [-Wimplicit-function-declaration]
  fcntl(remote_bitbang_fd, F_SETFL, O_NONBLOCK);
  ^~~~~
  fcvt
/Host/Work/openocd/openocd.git/src/jtag/drivers/remote_bitbang.c:63:27: error: 'F_SETFL' undeclared (first use in this function); did you mean 'FD_SET'?
  fcntl(remote_bitbang_fd, F_SETFL, O_NONBLOCK);
                           ^~~~~~~
                           FD_SET
/Host/Work/openocd/openocd.git/src/jtag/drivers/remote_bitbang.c:63:27: note: each undeclared identifier is reported only once for each function it appears in
/Host/Work/openocd/openocd.git/src/jtag/drivers/remote_bitbang.c:63:36: error: 'O_NONBLOCK' undeclared (first use in this function); did you mean 'SMTO_BLOCK'?
  fcntl(remote_bitbang_fd, F_SETFL, O_NONBLOCK);
                                    ^~~~~~~~~~
                                    SMTO_BLOCK
/Host/Work/openocd/openocd.git/src/jtag/drivers/remote_bitbang.c: In function 'remote_bitbang_rread':
/Host/Work/openocd/openocd.git/src/jtag/drivers/remote_bitbang.c:157:27: error: 'F_SETFL' undeclared (first use in this function); did you mean 'FD_SET'?
  fcntl(remote_bitbang_fd, F_SETFL, 0);
                           ^~~~~~~
                           FD_SET
make[2]: *** [Makefile:3248: src/jtag/drivers/src_jtag_drivers_libocdjtagdrivers_la-remote_bitbang.lo] Error 1
make[2]: Leaving directory '/tmp/build/win32/openocd'
make[1]: *** [Makefile:3794: all-recursive] Error 1
make[1]: Leaving directory '/tmp/build/win32/openocd'
make: *** [Makefile:1627: all] Error 2

```

I also tried a linux32 build but that didn't even get that far - the following is just the end of the output log:

```
bash ~/Downloads/openocd-build.git/scripts/build.sh --linux32

...

Initial shared libraries...
ls: cannot access /tmp/install/centos32/bin/*.dll: No such file or directory
/tmp/install/centos32/lib/:
total 576
drwxr-xr-x 3 root root   4096 Jan  4 10:11 cmake
-rw-r--r-- 1 root root 134578 Jan  4 10:11 libftdi1.a
lrwxrwxrwx 1 root root     13 Jan  4 10:11 libftdi1.so -> libftdi1.so.2
lrwxrwxrwx 1 root root     17 Jan  4 10:11 libftdi1.so.2 -> libftdi1.so.2.2.0
-rwxr-xr-x 1 root root 122280 Jan  4 10:11 libftdi1.so.2.2.0
-rw-r--r-- 1 root root 170304 Jan  4 10:11 libusb-1.0.a
-rwxr-xr-x 1 root root    970 Jan  4 10:11 libusb-1.0.la
lrwxrwxrwx 1 root root     19 Jan  4 10:11 libusb-1.0.so -> libusb-1.0.so.0.1.0
lrwxrwxrwx 1 root root     19 Jan  4 10:11 libusb-1.0.so.0 -> libusb-1.0.so.0.1.0
-rwxr-xr-x 1 root root 116216 Jan  4 10:11 libusb-1.0.so.0.1.0
-rw-r--r-- 1 root root  23274 Jan  4 10:11 libusb.a
-rwxr-xr-x 1 root root    962 Jan  4 10:11 libusb.la
drwxr-xr-x 2 root root   4096 Jan  4 10:11 pkgconfig

/tmp/install/centos32/lib/cmake:
total 4
drwxr-xr-x 2 root root 4096 Jan  4 10:11 libftdi1

/tmp/install/centos32/lib/cmake/libftdi1:
total 12
-rw-r--r-- 1 root root 2465 Jan  4 10:11 LibFTDI1Config.cmake
-rw-r--r-- 1 root root 1267 Jan  4 10:11 LibFTDI1ConfigVersion.cmake
-rw-r--r-- 1 root root  524 Nov 21  2014 UseLibFTDI1.cmake

/tmp/install/centos32/lib/pkgconfig:
total 16
-rw-r--r-- 1 root root 318 Jan  4 10:11 libftdi1.pc
-rw-r--r-- 1 root root 290 Jan  4 10:11 libftdipp1.pc
-rw-r--r-- 1 root root 339 Jan  4 10:11 libusb-1.0.pc
-rw-r--r-- 1 root root 309 Jan  4 10:11 libusb.pc
user@slinky:~/Downloads$ 
```

**Comment by TommyMurphyTM1234** [comment]
FWIW I deleted ~/Work/openocd and ~/Downloads/openocd-build.git and got the build scripts from scratch and tried a --linux32 build and the full log is attached.
[openocd-gme-build.log](https://github.com/gnu-mcu-eclipse/openocd-build/files/1603735/openocd-gme-build.log)

**Comment by ilg-ul** [comment]
Sorry, the repo was temporarily inconsistent. 

I updated it to my latest version and my full multi-platform build looks ok.

Please note that the CentOS Docker images are _almost_ final, I hope they'll be ready this weekend.

**Comment by TommyMurphyTM1234** [comment]
Ok thanks.
Let me try again and post back asap.

**Comment by TommyMurphyTM1234** [comment]
Just to confirm that --linux32 and --win32 builds work fine now.
I didn't try --linux64 and --win64 but I presume that they work too.
Thanks.

**Comment by TommyMurphyTM1234** [comment]
FWIW --win64 and --linux64 also built fine too.

**Comment by ilg-ul** [comment]
Thank you.
